Assets/Scripts/Stats/Health.cs

```diff
@@ -1,23 +1,31 @@
 using Assets.Scripts.Stats;
-using System;
 using System.Collections;
-using System.Collections.Generic;
+using Unity.Collections;
 using UnityEngine;
 using UnityEngine.Events;
 
-public class Health : Stat
+public class Health : RegeneratingStats
 {
-    [SerializeField] bool _isInvincible;
+    [SerializeField] float _regenRateInSeconds;
+    [SerializeField] Armor _armor;
```

Comment:
This creates strong coupling. Armor and health are separate components and thus should remain independent of each other. 
I suggest having an interface IDamageable that would be responsible for dealing with taking damage, and from there have a class DamageHandler that would implement the interface. From there, force me to use health and allow me to choose if the entity has armor or not.

Reply:
There is a new commit, implementing the interface IDamageable and the DamageHandler class